src/pkg-sqlite.c

```diff
@@ -737,9 +845,53 @@ pkg_sqlite_init ()
         NULL
     };
 
+    static sqlite3_vfs vfs_methods = {
+        3,
+        0,
+        0,
+        NULL,
+        "ldmud",
+        NULL,
+        sl_vfs_open,
+        sl_vfs_delete,
+        sl_vfs_access,
+        sl_vfs_full_pathname,
+
+        NULL,
+        NULL,
+        NULL,
+        NULL,
+
+        sl_vfs_randomness,
+        sl_vfs_sleep,
+        sl_vfs_current_time,
+        sl_vfs_get_last_error,
+        sl_vfs_current_time_int64,
+
+        NULL,
+        NULL,
+        NULL
+    };
+
+    sqlite3_vfs *orig_vfs;
+
     sqlite3_config(SQLITE_CONFIG_MALLOC, &mem_methods);
     sqlite3_config(SQLITE_CONFIG_LOG, &sl_log, NULL);
 
+    orig_vfs = sqlite3_vfs_find(NULL);
+    if (orig_vfs == NULL)
+        fatal("SQLite: No VFS registered.\n");
+
+
+    vfs_methods.szOsFile = orig_vfs->szOsFile;
+    vfs_methods.mxPathname = orig_vfs->mxPathname;
+    vfs_methods.pAppData = orig_vfs;
```

Comment:
Ist es eigentlich garantiert, dass das immer unter der Adresse liegt, also sqlite3 das nicht bewegt o.ae.?

Reply:
Also in der Dokumentation zu sqlite3_vfs_find() gibt es keine Hinweise über eine begrenzte Lebensdauer des Ergebnisses. (Praktisch verwendet sqlite3 genauso wie die Implementation im pkg statische Variablen für die VFS-Strukturen.)